net: lwm2m: Way to pass a destination hostname to socket

icatorze opened the pull request:
net: lwm2m: When mbedtls CONFIG_MBEDTLS_SERVER_NAME_INDICATION is enabled, a destination hostname must be passed to socket to properly connect do lwm2m server.

icatorze committed:
net: lwm2m: Way to pass a destination hostname to socket

net: lwm2m: When mbedtls CONFIG_MBEDTLS_SERVER_NAME_INDICATION is enabled, a destination hostname must be passed to socket to properly connect do lwm2m server.

icatorze committed:
net: lwm2m: Way to pass a destination hostname to socket

net: lwm2m: When mbedtls CONFIG_MBEDTLS_SERVER_NAME_INDICATION is enabled, a destination hostname must be passed to socket to properly connect do lwm2m server.

jukkar commented:
Please squash the commits together as the latter one is just changing stuff introduced in the first commit.

icatorze committed:
net: lwm2m: Way to pass a destination hostname to socket

net: lwm2m: When mbedtls CONFIG_MBEDTLS_SERVER_NAME_INDICATION is enabled,
a destination hostname must be passed to socket to properly connect do lwm2m
server.
Signed-off-by: Jair Jack <jack@icatorze.com.br>
